### ISSUE TITLE
Sync failure on Windows

### ISSUE THREAD

**Issue by altsheets** [ethereum/go-ethereum#2087]
I had it working in September, now I am trying to catch up.

But it just stops syncing by itself. I have already restarted it for the 4th time now, which seems to be the only way to continue importing blocks. 

    > web3.version
    {
      api: "0.13.0",
      client: "Geth/v1.2.2/windows/go1.4.2",
      ethereum: "62",
      network: "1", ...
    }

(windows, installed via chocolatey, see https://github.com/ethereum/go-ethereum/issues/2086#issuecomment-166142038)

Any log file that might help to report what is going wrong?

Is there a bootstrap file with the whole blockchain? 

Thanks

**Comment by altsheets** [comment]
Situation:

    I1221 14:24:18.661271    7188 downloader.go:275] Block synchronisation started
    I1221 14:24:55.187361    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 275 txs in 13.9577983s. #379037 [c5e3f881 / 6b859d53]
    I1221 14:25:06.790024    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 293 txs in 11.6006635s. #379293 [34d6c717 / ae21a025]
    I1221 14:25:28.770281    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 942 txs in 21.9752569s. #379549 [21509882 / a6d3af65]
    I1221 14:25:37.177762    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 257 txs in 8.4064808s. #379805 [0da6a593 / 89ef7a4b]
    I1221 14:26:39.046301    7188 chain_manager.go:722] imported 1 block(s) (0 queued 0 ignored) including 2 txs in 19.0011ms. #379806 [a3074375 / a3074375]
    I1221 14:26:39.656336    7188 chain_manager.go:722] imported 2 block(s) (0 queued 0 ignored) including 1 txs in 38.0022ms. #379808 [ad835f32 / 413b6c89]
    I1221 14:26:39.793344    7188 chain_manager.go:722] imported 4 block(s) (0 queued 0 ignored) including 0 txs in 77.0044ms. #379812 [5e2c1a26 / cab679d2]
    I1221 14:26:40.022357    7188 chain_manager.go:722] imported 8 block(s) (0 queued 0 ignored) including 15 txs in 209.0119ms. #379820 [022546aa / 7c17fd9d]
    I1221 14:26:43.037529    7188 chain_manager.go:722] imported 16 block(s) (0 queued 0 ignored) including 9 txs in 512.0293ms. #379836 [372dc5fc / 6d361b74]
    I1221 14:26:45.871691    7188 chain_manager.go:722] imported 32 block(s) (0 queued 0 ignored) including 32 txs in 1.0090577s. #379868 [f07b9ca1 / a9a1c59d]
    I1221 14:26:54.210168    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 271 txs in 8.3374768s. #380124 [aa0c8f5c / 64561d53]
    I1221 14:26:56.570303    7188 chain_manager.go:722] imported 64 block(s) (0 queued 0 ignored) including 81 txs in 2.3591349s. #380188 [020b5b2f / 4a4be187]
    I1221 14:27:04.476756    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 268 txs in 7.9054521s. #380444 [a7031f22 / a6095181]
    I1221 14:27:12.458212    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 222 txs in 7.9794564s. #380700 [141123b6 / 3422a01a]
    I1221 14:27:21.403724    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 250 txs in 8.9445116s. #380956 [cb5dcc44 / ef6668ef]
    I1221 14:27:31.568305    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 338 txs in 10.1635813s. #381212 [8649591f / 18db389e]
    I1221 14:27:44.923069    7188 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 301 txs in 13.3537638s. #381468 [2ed17b22 / 68a639ea]
    I1221 14:27:55.292662    7188 chain_manager.go:722] imported 234 block(s) (0 queued 0 ignored) including 340 txs in 10.368593s. #381702 [6aa6d206 / aa36e018]
    I1221 14:27:55.680684    7188 chain_manager.go:722] imported 0 block(s) (0 queued 256 ignored) including 0 txs in 387.0221ms. #381610 [9cc9053e / 4294216b]
    I1221 14:27:55.829693    7188 chain_manager.go:722] imported 0 block(s) (0 queued 92 ignored) including 0 txs in 148.0085ms. #381702 [506eb029 / aa36e018]

and now it is stuck there for the past 9 minutes already.

**Comment by altsheets** [comment]
and again ...

    I1221 14:51:17.346855    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 225 txs in 9.056518s. #399943 [5982bf4e / 8eaa8274]
    I1221 14:51:26.086355    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 289 txs in 8.7384998s. #400199 [5d9c3b91 / 34791cc1]
    I1221 14:51:33.730792    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 223 txs in 7.6434372s. #400455 [d5e3c85d / dfaf961a]
    I1221 14:51:39.605128    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 205 txs in 5.873336s. #400711 [2b2a7885 / 6942b449]
    I1221 14:51:45.900488    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 348 txs in 6.29536s. #400967 [f3f67b8e / 2ac6e7fc]
    I1221 14:51:53.004895    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 234 txs in 7.1034063s. #401223 [2a5648a7 / e787e4ae]
    I1221 14:52:00.304312    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 290 txs in 7.2984175s. #401479 [2e9689a3 / 239fb0e4]
    I1221 14:52:10.372888    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 838 txs in 10.0665758s. #401735 [3793e5b8 / 869808ef]
    I1221 14:52:16.713251    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 294 txs in 6.3393626s. #401991 [10f5cdc6 / 4bf94bf1]
    I1221 14:52:21.061499    8932 chain_manager.go:722] imported 192 block(s) (0 queued 0 ignored) including 163 txs in 4.3462486s. #402183 [b5d35052 / b2cfe41b]
    I1221 14:52:30.536041    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 341 txs in 9.4725418s. #402439 [a6f675d2 / 4de28534]
    I1221 14:52:38.811515    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 216 txs in 8.2744733s. #402695 [e6f3d34c / fa1bfb32]
    I1221 14:52:45.249883    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 283 txs in 6.4383682s. #402951 [9bd3dfcf / 047c5937]
    I1221 14:52:53.617361    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 236 txs in 8.3664785s. #403207 [fb9db22e / 669f105e]
    I1221 14:53:05.051015    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 357 txs in 11.4316539s. #403463 [856b4481 / 3d4b0606]
    I1221 14:53:12.242427    8932 chain_manager.go:722] imported 256 block(s) (0 queued 0 ignored) including 312 txs in 7.1914113s. #403719 [e3822522 / 9a9288fa]


this time stuck for 27 minutes now. 

I guess it's safe to say ... only kill-it-and-restart will help?

**Comment by altsheets** [comment]
Restarted three times already.

Man, 700k blocks? This will take weeks to sync ...

**Comment by fjl** [comment]
Please upgrade to geth 1.3 if possible.When catching up for the first time, you can use Fast Sync, which can be enabled using the --fast command line flag.

**Comment by altsheets** [comment]
Thanks a lot for your answer.
After restarting another 6 times, I could make it to block 532257 now.

> Please upgrade to geth 1.3 if possible.

Yes, I would like to.

Please you do the upgrade on chocolatey --> https://chocolatey.org/packages/geth-stable   
The newest version there is 1.2.2.0

Thanks.

**Comment by altsheets** [comment]
# Newest version - new method

... I [asked](https://twitter.com/altsheets/status/678999101309087744) tgerring (who had done the chocolatey versions up to 1.2.2.0), and got [this answer](https://twitter.com/TaylorGerring/status/679313455392927746):

    Manually download the build as listed on http://Ethereum.org 

So the *chocolatey install* is dead seemingly.
But that is where [google "ethereum windows"](https://www.google.de/search?q=ethereum+windows) brings you to.

## TODO:

I suggest to change the [wiki page](https://github.com/ethereum/go-ethereum/wiki/Installation-instructions-for-Windows).

**Comment by altsheets** [comment]
Update:

I had to restart 1.2.2.0 TWENTY times because it got stuck while syncing, again and again and again. Finally I gave up. 
I think you should issue a warning, that this version is broken beyond reasonable patience. It has cost me much time.

That is relevant because the 1.2.2.0 is the newest 'choco install geth-stable' 
which linked from 'ethereum wiki windows install' https://github.com/ethereum/go-ethereum/wiki/Installation-instructions-for-Windows 
which is the top 1 result for "google ethereum windows" https://www.google.de/search?q=ethereum+windows


Perhaps you should delete the whole chocolatey Geth package - or simply upgrade it to a newer version. Can't be too difficult. (I offer to admin the chocolatey repo in the future if you pay me)

Thanks.

:-)
	   

TL;DR: 
* version 1.2.2.0 is broken.
* So you need to take action and remove (or upgrade) the geth chocolatey package. 
* Your wiki is misleading. Google favors it.
* version 1.3.2 from https://ethereum.org/cli ... syncing is working flawlessly, in one go, without any restart from #696901 to now #719750. Looks like you fixed the bug.

**Comment by altsheets** [comment]
Perhaps I celebrated the 1.3.2 version too early ...

    I0103 17:48:36.407464    1412 blockchain.go:1230] imported 256 block(s) (0 queued 0 ignored) including 451 txs in 33.0168884s. #731014 [8cd55a94 / bf5d6600]
    I0103 17:49:09.858377    1412 blockchain.go:1230] imported 256 block(s) (0 queued 0 ignored) including 332 txs in 33.446913s. #731270 [ef8f75a7 / c2a664c1]
    I0103 17:49:38.696026    1412 blockchain.go:1230] imported 256 block(s) (0 queued 0 ignored) including 423 txs in 28.8286489s. #731526 [59260d61 / e12dd6fb]
    I0103 17:49:52.704828    1412 blockchain.go:1230] imported 134 block(s) (0 queued 0 ignored) including 217 txs in 14.0018009s. #731660 [45f1c865 / b44198c5]

**now it is 18:12 = 23 minutes later**, and nothing is happening anymore :-(


... trying to find out what is going on:

    geth attach
    
    > net.listening
    true
    > net.peerCount
    0

(what else could I check in such a moment to help you fix this annoying bug?)



... then I manually restarted geth, and it continues syncing. Now it looks like this:

    > net.peerCount
    5
    > net.peerCount
    5
    > net.peerCount
    10
    > net.peerCount
    12

Is the ethereum network itself so sparse - or the protocol so shaky - that a node can suddenly loose all his peers?

**Comment by tgerring** [comment]
The wiki has been updated to reflect latest Windows build information, including a not about the old version on Chocolatey

**Comment by altsheets** [comment]
Perfect. That should protect others to go through the same ordeal ;-)

**Comment by altsheets** [comment]
Now with 1.3.2 ... it took only *four* more restarts to sync the remaining ~60k blocks. Something is wrong there. What a strange software. But I could sync my node now. Finally. Whew. Thanks.

**Comment by altsheets** [comment]
> fjl changed the title ... to Sync failure on Windows 

Perfect. I had been wondering if I do exactly that. 
Perfect title, that nails it (and as I still had problems in 1.3.2 too ... it is not only a 1.2.2.0 problem.).

Thanks a lot. Enjoy your weekend ;-)

**Comment by karalabe** [comment]
Unless you can give us some logs, it's not something we can figure out. Please run geth with `--vmodule=downloader=6` and - if it gets stuck - paste the logs into a github gist so we might investigate.

**Comment by fjl** [comment]
I have identified the issue. It is the same issue as #1579. Will have a fix soon.

**Comment by altsheets** [comment]
>  run geth with --vmodule=downloader=6 

Thanks. Too late for now, but good to know for the future.

> Will have a fix soon.

Happy. Thanks.

**Comment by fjl** [comment]
This should be resolved as of geth 1.3.5.

**Comment by altsheets** [comment]
Congrats.   
Are there rewards?
